libcontainer/devices/devices_unix.go

```diff
@@ -94,6 +94,9 @@ func getDevices(path string) ([]*configs.Device, error) {
 			if err == ErrNotADevice {
 				continue
 			}
+			if os.IsNotExist(err) {
```

Comment:
If the path does not exist, it is clearly not a device.  As you say this is a racy situation, and it should anticpate the fact that devices can dissapear.  I believe the goal of the function is to list define the devices that are still there.  But of course the device could disappear even after it was defined as a device.

Either way I am fine with it. But version prevents other uses potentially hitting this error and not checking for ENOEXIST